### PR TITLE
Add sharding example

### PR DESCRIPTION
**Pull request by tianfeng92**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jobs:
               -sdk iphoneos \
               -configuration Debug \
               -derivedDataPath build
+            pushd .
```
Comment (FriggaHel) [inline .circleci/config.yml]: We should avoid using `pushd` and `cd`.
Replacing `cd build/Build/Products/Debug-iphoneos` by `pushd build/Build/Products/Debug-iphoneos` should achieve the same goal.

**Review by mhan83 — COMMENTED** [review]
So the secret sauce to sharding is that shell script to generate the test files? Should the README do more to describe what is happening, how it works, and how a user would incorporate that into their apps?

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ xcodebuild \
 ```shell
 APP=DemoApp/build/Build/Products/Debug-iphoneos/DemoApp.app \
 TEST_APP=DemoApp/build/Build/Products/Debug-iphoneos/DemoAppUITests-Runner.app \
+TEST_LIST_FILE=DemoApp/test_classes.txt \
```
Comment (alexplischke) [inline README.md:65]: It'd be good to explain where this file comes from. It's not immediately obvious that 1. there's a shell script as part of the build process that generates this file; 2. _what_ we are trying to demonstrate here by providing this file to saucectl; and 3. whether it's mandatory to have this at all times.

File: README.md [hunk 3]
```diff
@@ -51,11 +51,25 @@ APP=DemoApp/build/Build/Products/Debug-iphoneos/DemoApp.app \
...
 saucectl run
 ```
-
 ![sauce cloud example](assets/xcuitest.gif)
 
+### Run XCUITest in Parallel
+
+`saucectl` supports running tests in parallel by setting `shard` and `testListFile` in sauce config. Click [here](.sauce/sharding-config.yml) to check the details.
+
+There is a script to extract test classes in `DemoApp` project build phase. So when you're building `DemoApp` in this example repo, it also generates a `DemoApp/test_classes.txt`. This file can be used for sharding XCUITest by concurrency. You can also create your own `testListFile` on demand.
```
Comment (alexplischke) [inline README.md]:
```suggestion
There is a script available to extract test classes during the build phase of the DemoApp project. Consequently, when you build DemoApp using the provided repository as an example, it generates a `DemoApp/test_classes.txt` file. This file can be utilized for sharding XCUITest by concurrency. Additionally, you have the flexibility to create your own `testListFile` as needed.
```

**Review by mhan83 — APPROVED** [review]

**Review by alexplischke — APPROVED** [review]

**Review by FriggaHel — APPROVED** [review]

**Pull request merged**